mailbox/api/src/test/java/org/apache/james/mailbox/AbstractMailboxManagerTest.java

```diff
@@ -24,18 +24,22 @@
 import java.io.IOException;

```

Comment:
Ok, here I guess you will have troubles with unit tests.

This class is extended by every implementation... Even those that do not support Annotations (eg : JPA).

Either (1) you split conditions other annotations in a separate class, that will be extended only where you support storing annotations.

Or (2) add a capability, and check the capability is implemented at the beginning of the test.

Or (3) test it at the mapper level and write mocked tests on the StoreMailboxManager.

Or (4) exend your tests class where annotations are not supported and annotate them with @Ignore...

I prefer 1 and 3. I would be ok with 4.